presto-hive/src/main/java/com/facebook/presto/hive/InternalHiveSplit.java

```diff
@@ -40,74 +39,56 @@
     // Overhead of ImmutableList and ImmutableMap is not accounted because of its complexity.
     private static final int INSTANCE_SIZE = ClassLayout.parseClass(InternalHiveSplit.class).instanceSize() +
             ClassLayout.parseClass(String.class).instanceSize() +
-            ClassLayout.parseClass(Properties.class).instanceSize() +
             ClassLayout.parseClass(String.class).instanceSize() +
             ClassLayout.parseClass(OptionalInt.class).instanceSize();
-    private static final int INTEGER_INSTANCE_SIZE = ClassLayout.parseClass(Integer.class).instanceSize();
 
     private final String path;
     private final long end;
     private final long fileSize;
-    private final Properties schema;
-    private final List<HivePartitionKey> partitionKeys;
     private final List<InternalHiveBlock> blocks;
-    private final String partitionName;
     private final OptionalInt readBucketNumber;
     private final OptionalInt tableBucketNumber;
     private final boolean splittable;
```

Comment:
no we can't because when we create the path we call HiveUtil.isSplittable() which takes the file path.  There's no explicit guarantee that it's the same for every file because it's implemented by the FileInputFormat.

Reply:
Just checked it. You are right. It is weird though. 

Anyhow, it's just a single boolean. Not a big deal.